src/lib/transactions.ts

```diff
@@ -87,10 +93,32 @@ const processTransaction = (transaction: GQLEdgeInterface, type: string) => ({
   date: ""
 });
 
+export async function checkTransactionError(
+  transaction: GQLEdgeInterface | Transaction
+): Promise<boolean> {
+  if (transaction.node.owner.address !== AGENT_TOKEN_ADDRESS) {
+    return false;
+  }
+
+  return retryWithDelay(async (attempt) => {
```

Comment:
I think this might not always return a true or false and sometimes can return an error as well. So, i think we should handle the error cases as well.

Here is one case where there is an error with a gql query so we must handle error cases as well and maybe return false maybe in case of error:

```JSON
{"errors":[{"message":"query timed out"}],"data":null}

```

Reply:
@pawanpaudel93 is this error message under `data.data.transactions.edges`? looking for ways of filtering it out.

Reply:
Yes, this is just an example of an error i got for print transactions with timeout error and it is returned from the gql function.
```
{"errors":[{"message":"query timed out"}],"data":null}
```

`data.data.transactions.edges.length > 0` works perfectly fine. I was thinking to add error handling when both retries failed with graphql or we get the timeout error.

Reply:
I was thinking of modifying to this:

```typescript
export async function checkTransactionError(
  transaction: GQLEdgeInterface | Transaction
): Promise<boolean> {
  if (transaction.node.owner.address !== AGENT_TOKEN_ADDRESS) {
    return false;
  }

  return retryWithDelay(async (attempt) => {
    const data = await gql(
      AF_ERROR_QUERY,
      { messageId: transaction.node.id },
      txHistoryGateways[attempt % txHistoryGateways.length]
    );
    
    if (data?.data === null && (data as any)?.errors?.length > 0) {
      throw new Error(
        (data as any)?.errors?.[0]?.message || "GraphQL Error"
      );
    }
                
    return data.data.transactions.edges.length > 0;
  }, 2).catch(() => false);
}
```

Reply:
sounds good